cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -77,6 +83,91 @@ def _validate_device_specification(proto: v2.device_pb2.DeviceSpecification) ->
                     )
 
 
+def _build_gateset_and_gate_durations(
+    proto: v2.device_pb2.DeviceSpecification,
+) -> Tuple[cirq.Gateset, Dict[cirq.GateFamily, cirq.Duration]]:
+    gates_list: List[Union[Type[cirq.Gate], cirq.Gate, cirq.GateFamily]] = []
+    fsim_gates: List[Union[Type[POSSIBLE_FSIM_GATES], POSSIBLE_FSIM_GATES]] = []
+    gate_durations: Dict[cirq.GateFamily, cirq.Duration] = {}
+
+    # TODO(#5050) Describe how to add/remove gates.
+
+    for gate_spec in proto.valid_gates:
+        gate_name = gate_spec.WhichOneof('gate')
+        cirq_gates: List[Union[Type[cirq.Gate], cirq.Gate, cirq.GateFamily]] = []
+
+        if gate_name == 'syc':
+            cirq_gates = [ops.SYC]
+            fsim_gates.append(ops.SYC)
+        elif gate_name == 'sqrt_iswap':
+            cirq_gates = [cirq.SQRT_ISWAP]
+            fsim_gates.append(cirq.SQRT_ISWAP)
+        elif gate_name == 'sqrt_iswap_inv':
+            cirq_gates = [cirq.SQRT_ISWAP_INV]
+            fsim_gates.append(cirq.SQRT_ISWAP_INV)
+        elif gate_name == 'cz':
+            cirq_gates = [cirq.CZ]
+            fsim_gates.append(cirq.CZ)
+        elif gate_name == 'phased_xz':
+            cirq_gates = [
+                cirq.PhasedXZGate,
+                cirq.XPowGate,
+                cirq.YPowGate,
+                cirq.ZPowGate,
```

Comment:
Should we add a `cirq.ZPowGate` gate here given that we have special identifiers for `virtual_zpow` and `physical_zpow` ? Note that if `cirq.GateFamily(cirq.ZPowGate)` is present in the gateset, it will accept all (i.e. both tagged and untagged) instances of ZPowGate; in which case the `physical_zpow` and `virtual_zpow` gate families will be irrelevant.  

For example: What happens if the proto specification includes `phased_xz` and doesn't contain `virtual_zpow` ? Should a `cirq.Z(q)` be accepted or not? What if it contains `phased_xz` and `physical_zpow` but doesn't contain `virtual_zpow` ?